examples/sediment_meander_2d/meander_example.py

```diff
@@ -0,0 +1,116 @@
+
+# define function spaces
+DG_2d = FunctionSpace(mesh2d, 'DG', 1)
+vector_dg = VectorFunctionSpace(mesh2d, 'DG', 1)
+
+# choose directory to output results
+outputdir = 'outputs'
+
+morfac = 50
+dt = 2
+end_time = 5*3600
+viscosity_hydro = Constant(5*10**(-2))
+
+if os.getenv('THETIS_REGRESSION_TEST') is not None:
+    # the example is being run as a test
+    # run the spin-up by importing it
+    import meander_hydro  # NOQA
+    end_time = 3600.
+
+# initialise velocity and elevation
+with DumbCheckpoint("hydrodynamics_meander/elevation", mode=FILE_READ) as chk:
+    elev = Function(DG_2d, name="elevation")
+    chk.load(elev)
+    chk.close()
+with DumbCheckpoint('hydrodynamics_meander/velocity', mode=FILE_READ) as chk:
+    uv = Function(vector_dg, name="velocity")
+    chk.load(uv)
+    chk.close()
+
+# set up solver
+solver_obj = solver2d.FlowSolver2d(mesh2d, bathymetry_2d)
+options = solver_obj.options
+
+# this test case only uses bedload transport but using all slope effect corrections and secondary current
+options.sediment_model_options.solve_suspended_sediment = False
+options.sediment_model_options.use_bedload = True
+options.sediment_model_options.solve_exner = True
+options.sediment_model_options.use_angle_correction = True
+options.sediment_model_options.use_slope_mag_correction = True
+options.sediment_model_options.use_secondary_current = True
+options.sediment_model_options.use_advective_velocity_correction = False
+options.sediment_model_options.morphological_viscosity = Constant(1e-6)
+options.sediment_model_options.average_sediment_size = Constant(10**(-3))
+options.sediment_model_options.bed_reference_height = Constant(0.003)
+options.sediment_model_options.morphological_acceleration_factor = Constant(morfac)
+
+options.simulation_end_time = end_time/morfac
+options.simulation_export_time = options.simulation_end_time/45
+options.output_directory = outputdir
+options.check_volume_conservation_2d = True
+options.fields_to_export = ['uv_2d', 'elev_2d', 'bathymetry_2d']
+# using nikuradse friction
+options.nikuradse_bed_roughness = Constant(3*options.sediment_model_options.average_sediment_size)
+# set horizontal viscosity parameter
+options.horizontal_viscosity = Constant(viscosity_hydro)
+# crank-nicholson used to integrate in time system of ODEs resulting from application of galerkin FEM
+options.timestepper_type = 'CrankNicolson'
+options.timestepper_options.implicitness_theta = 1.0
+if not hasattr(options.timestepper_options, 'use_automatic_timestep'):
+    options.timestep = dt
+
+left_bnd_id = 1
+right_bnd_id = 2
+
+# set boundary conditions
+gradient_flux = (-0.053 + 0.02)/6000
+gradient_flux2 = (-0.02+0.053)/(18000-6000)
+gradient_elev = (10.04414-9.9955)/6000
+gradient_elev2 = (9.9955-10.04414)/(18000-6000)
+elev_init_const = (-max(bathymetry_2d.dat.data[:]) + 0.05436)
+flux_constant = Constant(-0.02)
+elev_constant = Constant(elev_init_const)
+
+swe_bnd = {}
+swe_bnd[3] = {'un': Constant(0.0)}
```

Comment:
There is a slight difference in the formulation in the equations if you enforce a zero boundary condition rather than not stating anything. This test case has issues at the boundaries (see section 5.2.1 of the paper https://www.sciencedirect.com/science/article/pii/S0098300420306324?casa_token=yW9-p0xX-WIAAAAA:pb6fwk5QcNZf6KXF0q6Ol451HIN6dFo-0kLQUrzB8z0tdhuBcfUz6lmHrPWaVJM5aqARHH-LXTA3#sec5) and I found that enforcing the strong boundary condition helped with this issue a bit.

Reply:
I think we can revisit the boundary conditions once bendy elements work